server/src/server.ts

```diff
@@ -144,6 +144,11 @@ connection.onRequest('bitbake/getVar', async (params: { variable: string, recipe
   return scanResult?.symbols.find(symbolInfo => symbolInfo.name === params.variable)?.finalValue
 })
 
+connection.onRequest(RequestMethod.IsSymbolDefinedInRecipe, async (params: RequestParams['IsSymbolDefinedInRecipe']) => {

```

Comment:
I really don't like doing so many request for every variable from the server to the client.
Scan results should remain in the server. So far we only need very specific variables at rare times in the client and it should stay that way for performance.

You should implement the feature "add-common-directories-to-variable-expansion-suggestions" in a different way. I'd prefer a basic static array over a dynamic recipe scan.

Reply:
I could gather all the variable names and make a single call to test which ones are truly defined.

In any case, I don't expect the performance overhead to be that important, since both the client and the server are running on the same machine.

Otherwise, the static version is already merged and working.

Reply:
I prefer keeping the code simple and go with the static approach. This code is useful but I don't think the complexity it adds and performance overhead is worth it.